docs/syntax/Cadence.syntax

```diff
@@ -0,0 +1,1482 @@
+    ;
+
+varVariableDeclaration:
+        access varKeyword nonReservedIdentifier *1variableDeclarationTypeAnnotation transfer expression *1variableDeclarationTrailingTransferExpression
+    ;
+
+variableDeclarationTypeAnnotation:
+        ":" typeAnnotation
+    ;
+
+variableDeclarationTrailingTransferExpression:
+        transfer expression
+    ;
+
+assignment:
+        assignmentTarget transfer assignmentValue
+    ;
+
+assignmentTarget:
+        expression
+    ;
+
+assignmentValue:
+        expression
+    ;
+
+swap:
+        swapExpressionLeft "<->" swapExpressionRight
+    ;
+
+swapExpressionLeft:
+        expression
+    ;
+
+swapExpressionRight:
+        expression
+    ;
+
+transfer:
+        transferEqual
+    |   transferMoveForced
+    |   transferMove
+    ;
+
+transferEqual:
+        "="
+    ;
+
+transferMoveForced:
+        moveForced
+    ;
+
+transferMove:
+        move
+    ;
+
+expression:
+        conditionalExpression  // 20
+    ;
+
+conditionalExpression:
+        orExpression *1conditionalExpressionBranches
+    ;
+
+conditionalExpressionBranches:
+        "?" conditionalExpressionBranchesTrue ":" conditionalExpressionBranchesFalse
+    ;
+
+conditionalExpressionBranchesTrue:
+        expression
+    ;
+
+conditionalExpressionBranchesFalse:
+        expression
+    ;
+
+orExpression:  // 30 '||'
+        1*andExpression( "||" )
+    ;
+
+andExpression:  // 40
+        1*comparisonExpression( "&&" )
+    ;
+
+comparisonExpression:  // 50
+        1*2nilCoalescingExpression( comparisonOp )
+    ;
+
+nilCoalescingExpression:  // 60
+        1*bitwiseOrExpression( nilCoalescingOperator )
+    ;
+
+nilCoalescingOperator:
+        "?" . "?"
+    ;
+
+bitwiseOrExpression:  // 70
+        1*bitwiseXorExpression( "|" )
+    ;
+
+bitwiseXorExpression:  // 80
+        1*bitwiseAndExpression( "^" )
+    ;
+
+bitwiseAndExpression:  // 90
+        1*bitwiseShiftExpression( "&" )
+    ;
+
+bitwiseShiftExpression:  // 100
+        1*additiveExpression( bitwiseShiftOp )
+    ;
+
+additiveExpression:  // 110
+        1*multiplicativeExpression( additiveOp )
+    ;
+
+multiplicativeExpression:  // 120
+        1*moveExpression( multiplicativeOp )
+    ;
+
+moveExpression:  // 130
+        *1moveOp castingExpression
+    ;
+
+castingExpression:  // 140
+        prefixExpression *castingExpressionCast
+    ;
+
+castingExpressionCast:
+        castingOp typeAnnotation
+    ;
+
+prefixExpression:  // 150
+        *1prefixOp postfixExpression
+    ;
+
+postfixExpression:  // 160
+        primaryExpression
+    ;
+
+forceUnwrapOp:
+        \ "!"
+    ;
+
+primaryExpression:
+        nullDenotation *leftDenotation
+    ;
+
+nullDenotation:
+        identifierExpression
+    ;
+
+leftDenotation:
+        forceUnwrapOp
+    |   accessExpression
+    |   invocationExpression
+    ;
+
+accessExpression:  // 170
+        memberExpression
+    |   indexExpression
+    ;
```

Comment:
member expression, indexing expression, and invocation expression are all part of the same precedence and should probably be in the same rule. As noted above, force unwrapping has lower precedence (needs to be in a "higher" rule)